New proposal step 2

Hidde-Heijnen opened the pull request:
Finished step 2 of create proposal

BenjaminMeijs commented on src/pages/NewProposal.tsx:
```diff
@@ -58,7 +58,7 @@ export const useNewProposalFormContext = () =>
 
 export type NewProposalFormProviderProps = {
-  children: any;
+  children: any;w
```
I don't think this 'w' is meant to be here.
Did you know VScode has an option to view what you've staged and what you're about to commit? I think github desktop must have something similar :).

Hidde-Heijnen replied:
the w isn't supposed to be there, and the date time picker is temporary. shadcn's new version contains a date picker, so I was thinking on waiting on that one, this will be the MVP of the date picker

BenjaminMeijs reviewed (approved):
As we discussed, some changes are needed. However, it is best to put these in a new PR

Hidde-Heijnen commented:
Things that got added in this branch:

+ date-utils.ts (All date functionality needed currently)
+ use-VotingSettings hook for retrieving minDuration
+ TimeZoneSelector with all the timezone options
+ Select component
+ Radio Group Card in RadioGroup
+ Labelled Input component

**Step two**
Step two styling completely updated and revamped. Seperate date, time and timezone pickers.
Step two takes into account the minDuration of a proposal of the dao with all the input options of the start and enddate.

Changes: 
- Changed "Links and resources" to "Resources" and fixed typo
- TextareaWYSIWYG isRequired error fix
- Updated viewStep2 with the new Step2Data

Hidde-Heijnen reviewed (commented):
Checked all changed files

BenjaminMeijs reviewed (approved):
same as before
